nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -175,16 +191,16 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
 
   double linear_vel, angular_vel;
 
-  // Find distance^2 to look ahead point (carrot) in robot base frame
-  // This is the chord length of the circle
-  const double carrot_dist2 =
-    (carrot_pose.pose.position.x * carrot_pose.pose.position.x) +
-    (carrot_pose.pose.position.y * carrot_pose.pose.position.y);
+  double desired_curvature = calculateCurvature(carrot_pose.pose.position);
```

Comment:
I think this is just "path_curvature" or simply "curvature". Desired seems like an odd word choice

Reply:
I want to disambiguate between the curvature that is used for regulating velocity, and the curvature that is used for calculating the angular velocity ("desired curvature")

Reply:
Maybe I could rename this to `command_curvature`?

Reply:
Lookahead curvature perhaps

Reply:
Well, but the curvature used for computing the commanded angular vel is also a lookahead curvature.

Reply:
On L228 that is the current variable named `desired_curvature` which we're looking at renaming, so that seems to line up perfectly :wink:

Reply:
`lookahead_curvature`

Reply:
For which one? There are two different curvatures at play here and they are both lookahead curvatures.

Reply:
`regulation_curvature` is used by the regulation term for curvature, so I think this is great.

`desired_curvature` is the actual lookahead-point's curvature that is being used to generate velocities and perform collision checking, so I think that should be `lookahead_curvature`. 

The regulation one's parameter implies "lookahead distance" but it is not itself the actual Pure Pursuit lookahead point used by the algorithm. 

Does that make sense? The only change I'm asking for is `desired_curvature` -> `lookahead_curvature`. I know I'll read  `desired` in the future and always wonder what that means in particular: Desired by _whom_, given there are multiple consumers.

Reply:
Ok that makes sense. I will change it to that!